File: shared/types/PersonaContext.ts
import type { TriggerEvent } from './TriggerEvent';

export type PersonaMode = 'friend' | 'commander';

export interface PersonaContext {
  mode: PersonaMode;
  energyPulse: number;
  triggerEvent: TriggerEvent;
  streakDay: number;
  userName: string;
  activeModule?: string;
}

export interface PersonaResponse {
  message: string;
  mode: PersonaMode;
  triggerEvent: TriggerEvent;
}

File: apple/Bennett/Models/PersonaContext.swift
import Foundation

enum PersonaMode: String, Codable, Sendable {
    case friend
    case commander
}

struct PersonaContext: Codable, Sendable {
    var mode: PersonaMode
    var energyPulse: Int
    var triggerEvent: TriggerEvent
    var streakDay: Int
    var userName: String
    var activeModule: String?
}

struct PersonaResponse: Codable, Sendable {
    let message: String
    let mode: PersonaMode
    let triggerEvent: TriggerEvent
}
